pom.xml

```diff
@@ -93,7 +93,7 @@
     <maven-enforcer-plugin.version>3.0.0-M1</maven-enforcer-plugin.version>
     <maven-docker-plugin.version>0.42.1</maven-docker-plugin.version>
 
-    <java.version>1.8</java.version>
+    <java.version>8</java.version>
```

Comment:
Does this mean that the Java 17 compiler still generates the byte code that targets Java 8 runtime, which is compatible with Java 17 runtime?  Does this prevent any optimization from Java 17?

Reply:
This PR aims to make Hudi compilable with Java 17, and the default jdk would remain to be JDK8. This property can be overriden with `-Djava.version=17` and I've added new CI checks with the override

Reply:
Got it.  Could you file a JIRA on making potential changes to the artifact building with `-Djava.version=17` in `scripts/release/deploy_staging_jars.sh`?  We'll discuss what Spark bundles will use Java 17 build going forward.